backend/app/services/dashboard_legs.py

```
@@ -134,21 +148,108 @@ def format_decision_reason(
+    The window is intentionally inclusive on both ends.
+    """
+    if dte <= 0:
+        return False
+    earnings_iso = earnings_lookup(ticker)
+    if not earnings_iso:
+        return False
+    try:
+        earnings_date = date.fromisoformat(str(earnings_iso)[:10])
+    except (TypeError, ValueError):
+        return False
+    today_d = date.today()
```

Ruff (0.15.12)

[warning] `datetime.date.today()` used

(DTZ011)